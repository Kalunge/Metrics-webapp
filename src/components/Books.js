import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import { FaBible } from 'react-icons/fa';
import { authenticate, getBooks } from '../redux/books/books';
// import bible from '../img/bible.jpg';
import Book from './Book';
import Header from './Header';

const Books = () => {
  const { books } = useSelector((state) => state.books);
  const { userInfo } = useSelector((state) => state.books);
  const dispatch = useDispatch();

  useEffect(() => {
    if (!userInfo.token) {
      dispatch(authenticate());
    }
    if (!books.length) {
      dispatch(getBooks());
    }
  }, [dispatch]);

  const oldtestament = books
    .filter((book) => book.author === 'Mois√©s')
    .slice(0, 5);

  return (
    <>
      <Header metric="Old Testament" />
      <div className="row">
        <div className="col-7">
          <FaBible size={98} className="fa-icon" />
        </div>
        <div className="col-5 mt-5 text">
          <h1 className="d-block">Holy bible </h1>
          <p className="btn btn-dark">66 Books</p>
        </div>
        <h5 className="text mt-2">Books by Moses</h5>
        {oldtestament.map((book) => (
          <Book key={book.abbrev.en} book={book} />
        ))}
      </div>
    </>
  );
};

export default Books;
