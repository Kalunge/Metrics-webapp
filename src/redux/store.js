import { createStore, applyMiddleware, combineReducers } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import booksReducer from './books/books';

const middleware = [thunk, logger];

const reducer = combineReducers({
  books: booksReducer,
});
const store = createStore(reducer, applyMiddleware(...middleware));

export default store;
