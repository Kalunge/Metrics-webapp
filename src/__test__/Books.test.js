import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/store';
import Books from '../components/Books';

it('renders   Books correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Books />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
