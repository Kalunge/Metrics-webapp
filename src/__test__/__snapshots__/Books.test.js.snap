// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders   Books correctly 1`] = `
Array [
  <div
    className="navBar mt-3 mb-3"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      <svg
        color="pink"
        fill="currentColor"
        height={42}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "pink",
          }
        }
        viewBox="0 0 512 512"
        width={42}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
        />
      </svg>
    </button>
    <h2>
      Old Testament
       
    </h2>
    <svg
      color="pink"
      fill="currentColor"
      height={42}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": "pink",
        }
      }
      viewBox="0 0 24 24"
      width={42}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M12 15c1.66 0 2.99-1.34 2.99-3L15 6c0-1.66-1.34-3-3-3S9 4.34 9 6v6c0 1.66 1.34 3 3 3zm5.3-3c0 3-2.54 5.1-5.3 5.1S6.7 15 6.7 12H5c0 3.42 2.72 6.23 6 6.72V22h2v-3.28c3.28-.48 6-3.3 6-6.72h-1.7z"
      />
    </svg>
    <svg
      color="pink"
      fill="currentColor"
      height={42}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": "pink",
        }
      }
      viewBox="0 0 24 24"
      width={42}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58a.49.49 0 00.12-.61l-1.92-3.32a.488.488 0 00-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54a.484.484 0 00-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58a.49.49 0 00-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
      />
    </svg>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-7"
    >
      <svg
        className="fa-icon"
        fill="currentColor"
        height={98}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 448 512"
        width={98}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M448 358.4V25.6c0-16-9.6-25.6-25.6-25.6H96C41.6 0 0 41.6 0 96v320c0 54.4 41.6 96 96 96h326.4c12.8 0 25.6-9.6 25.6-25.6v-16c0-6.4-3.2-12.8-9.6-19.2-3.2-16-3.2-60.8 0-73.6 6.4-3.2 9.6-9.6 9.6-19.2zM144 144c0-8.84 7.16-16 16-16h48V80c0-8.84 7.16-16 16-16h32c8.84 0 16 7.16 16 16v48h48c8.84 0 16 7.16 16 16v32c0 8.84-7.16 16-16 16h-48v112c0 8.84-7.16 16-16 16h-32c-8.84 0-16-7.16-16-16V192h-48c-8.84 0-16-7.16-16-16v-32zm236.8 304H96c-19.2 0-32-12.8-32-32s16-32 32-32h284.8v64z"
        />
      </svg>
    </div>
    <div
      className="col-5 mt-5 text"
    >
      <h1
        className="d-block"
      >
        Holy bible 
      </h1>
      <p
        className="btn btn-dark"
      >
        0
        Books
      </p>
    </div>
    <h5
      className="text mt-2"
    >
      Books by Moses
    </h5>
  </div>,
]
`;
